vm/src/vm.rs

```diff
@@ -107,6 +107,16 @@ impl VirtualMachine {
         self.new_exception(value_error, msg)
     }
 
+    pub fn new_key_error(&mut self, msg: String) -> PyObjectRef {
```

Comment:
Instead of `String` the preferred type is more `&str` since this will allow you to pass string arguments with the `.to_string()` suffix. This is a minor detail though.

Reply:
I was just trying to match the other new_xxx_error functions. I can go ahead and change them all to take a string slice if that's better.

Reply:
Yes, this is very good. Leave it like this, it can be done later, it is not important now.